### PR TITLE
authorized_key: support --diff

### PR DESCRIPTION
**Pull request by mgedmin**
### Test plan

``` yml
# test.yml

---
- hosts: localhost
  gather_facts: no
  vars:
    key: "{{ lookup('file', '~/.ssh/id_rsa.pub') }}"
    user: "{{ ansible_user }}"
  tasks:
    - authorized_key: path=./auth_keys key="{{ key }}" user="{{ user }}"
```

```
$ ansible-playbook -i localhost, test.yml -CD
$ ansible-playbook -i localhost, test.yml -D
```
### Expected output

```
PLAY [localhost] ***************************************************************

TASK [authorized_key] **********************************************************
--- before: ./auth_keys
+++ after: ./auth_keys
@@ -0,0 +1 @@
+ssh-rsa AAAA.....

changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0   
```

### CODE REVIEW

**Comment by mgedmin** [comment]
Migrated from https://github.com/ansible/ansible-modules-core/pull/3026

**Comment by edgarsj** [comment]
+1. Was really expecting this to work already when using authorized_keys module.

**Comment by robinro** [comment]
This is maybe a good occasion to also add a testcase for checkmode and the diff?

Something like 
```
- authorized_key: key={{foo}} ...
  check_mode: True
  register: add_key

- assert: 
    that: 
      - 'add_key|changed'
      - '"part-of-key" in add_key.diff...'```

**Comment by mgedmin** [comment]
Rebased on devel and added an integration test.

(Note to self: run just this one single test with `make integration TARGET=non_destructive TEST_FLAGS='-t test_authorized_key'`.)

File: lib/ansible/modules/system/authorized_key.py [hunk 1]
```diff
@@ -390,14 +390,20 @@ def parsekey(module, raw_key, rank=None):
 
     return (key, key_type, options, comment, rank)
```
Comment (gundalow) [inline lib/ansible/modules/system/authorized_key.py:391]: The file mode has changed, please change it back to 644

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

File: lib/ansible/modules/system/authorized_key.py (hunk 1 above)
Comment (mgedmin) [reply lib/ansible/modules/system/authorized_key.py:391]: Whoops, vim autocommand magic strikes again!

**Comment by mgedmin** [comment]
Rebased and force-pushed to fix the accidental chmod.

**Review by gundalow — APPROVED** [review]
shipit 

Bonus points for test case. I'd like someone else to look at this before merging.

**Review by robinro — APPROVED** [review]
LGTM

**Pull request merged**